src/screens/token/components/account-settings/account-settings.tsx

```diff
@@ -49,33 +49,43 @@ export class AccountSettingsComponent extends React.Component<IProps & IExternal
             title: translate('AccountSettings.manageAccount'),
             key: '',
             showSecurityWarning: false,
-            isLoading: false
+            isLoading: false,
+            isModalVisible: props.isVisible
         };
     }
 
-    public async revealPrivateKey() {
+    public componentDidUpdate(prevProps: IExternalProps) {
+        if (this.props.isVisible !== prevProps.isVisible) {
+            this.setState({ isModalVisible: this.props.isVisible });
+        }
+    }
+
+    public revealPrivateKey() {
         try {
-            // TODO: not working because of modal over modal
-            const password = await PasswordModal.getPassword();
+            this.setState({ isModalVisible: false }, async () => {
+                const password = await PasswordModal.getPassword();
```

Comment:
i think we could have the same problems here, as at this moment the account options modal is not completely closed. A safe method would be to open password modal when onModalHide is called on account options modal.